### PR TITLE
[MASTER]:Hotfix/conference 2018

### LINKED ISSUE #1811: 2018 Conference: full conference program update
Full build out of site with program information for each day.
Content TBD

### PR DESCRIPTION
**Pull request by chrisgarrity**
### Resolves:

Resolves #1811 Thanks @shickey

### CODE REVIEW

File: src/routes.json [hunk 1]
```diff
@@ -58,25 +44,39 @@
...
+        "pattern": "^/conference/2018/:id/details/?$",
         "routeAlias": "/conference(?!/201[4-5])",
-        "view": "conference/2016/index/index",
+        "view": "conference/2018/details/details",
+        "title": "Event Details"
+    },
+    {
+        "name": "conference-expectations-2018",
+        "pattern": "^/conference/2018/expect/?$",
+        "routeAlias": "/conference(?!/201[4-5])",
```
Comment (rschamp) [inline src/routes.json]: We should update these aliases to `/conference(?!/20\d\d)`

File: src/views/conference/2018/details/details.jsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+require('./details.scss');
+
+class ConferenceDetails extends React.Component {
+    componentDidMount () {
+        let pathname = window.location.pathname.toLowerCase();
+        if (pathname[pathname.length - 1] === '/') {
+            pathname = pathname.substring(0, pathname.length - 1);
+        }
+        const path = pathname.split('/');
+        const detailsId = path[path.length - 2];
```
Comment (rschamp) [inline src/views/conference/2018/details/details.jsx]: Just a suggestion (in the future?): `window.location.pathname.toLowerCase().split('/').filter(Boolean)[2]`

File: src/views/conference/2018/schedule/schedule.jsx [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+            'handleScheduleChange',
+            'renderChunkItems'
+        ]);
+    }
+    componentDidMount () {
+        const day = window.location.hash.substr(1) || 'thursday';
+        this.handleScheduleChange(day);
+    }
+    handleScheduleChange (day) {
+        window.history.replaceState(history.state, '', `#${day}`);
```
Comment (rschamp) [inline src/views/conference/2018/schedule/schedule.jsx]: This is probably copied from previous years, but kind of odd we're using `replaceState` with hashes when we could be using real URLs.

**Comment by chrisgarrity** [comment]
@rschamp yes, I asked about the localization too. The conference committee decided that since the conference is all in English it would be fine for the site not to be localized. They didn't want to set any expectations.

**Pull request merged**